include/parallel/parallel.h

```diff
@@ -1098,6 +1098,14 @@ class Communicator
                      T send,
                      std::vector<T> & recv) const;
 
+   /**
+    * Gather overload for string types
+    */
+   template <typename T>
+   inline void gather(const unsigned int root_id,
+                      std::basic_string<T> send,
```

Comment:
Should we be adding an overload so we can pass send via const reference?  I suppose if we're about to communicate N_proc*O(N) data then a single in-memory copy of O(N) data isn't so bad...